src/main/scala/scalismo/faces/image/filter/Median.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package scalismo.faces.image.filter
+
+import scalismo.faces.image.PixelImage
+
+/** median image filter */
+object Median {
+  // extract median value
+  private def median(values: Seq[Double]): Double = {
+    if(values.length <= 1)
+      values.head
```

Comment:
This line will throw for an empty sequence. Should we cover this case with a `require`?